main.py

```diff
@@ -67,6 +70,8 @@ def main():
     view = View(mymap=osmMap, simulation=sim, window_size=windowSize)
     app = Controller(model=sim, view=view)
     app.main_loop()
+    if sim.pathfindFile != None:
+        sim.pathfindFile.close()
```

Comment:
The second answer in this stack overflow question might help you with that: (use the __del__ or __exit__ property of an object).

https://stackoverflow.com/questions/865115/how-do-i-correctly-clean-up-a-python-object